### PR TITLE
chore: standardize buttons using a button component

### PR DESCRIPTION
**Pull request by MarioRodrigues10**
chore: use a Button component

### CODE REVIEW

**Comment by diogogmatos** [comment]
[Standardize buttons using a button component](https://trello.com/c/m4kxN5gc/8-standardize-buttons-using-a-button-component)

File: pages/register/[uuid].js [hunk 1]
```diff
@@ -89,7 +89,7 @@ function Register() {
             onChange={(e) => updatePasswordConfirmation(e.currentTarget.value)}
           />
           <Button
-            type="submit"
+            title="submit"
```
Comment (joaodiaslobo) [inline pages/register/[uuid].js:92]: This doesn't look quite right 🤔
image

**Review by joaodiaslobo — CHANGES_REQUESTED** [review]

File: components/Button/index.tsx [hunk 2]
```diff
@@ -1,16 +1,20 @@
 import { ButtonHTMLAttributes } from "react";
 
 interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
-  text: string;
+  title: string;
+  description?: string;
```
Comment (ruilopesm) [inline components/Button/index.tsx:5]: Why does a button have a description? 🤔

Comment (MarioRodrigues10) [reply components/Button/index.tsx:5]: There's a lot of buttons who has like a second title or a subtitle, for example in store the button has a title "Redeem" with a subtitle "X tokens".

**Review by tiago-bacelar — COMMENTED** [review]
I feel like the title field could be a child component. I know 99% of the time it's text, but for the other 1% it would be really useful. Not to mention, it would be less verbose, more intuitive and better follow the composition model of react.

File: components/Button/index.tsx [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
 
 export default function Button({
-  text,
+  title,
+  description,
   type,
   disabled,
   onClick,
   customStyle,
+  bold,
```
Comment (Darguima) [inline components/Button/index.tsx:17]:
```suggestion
  bold,
  ...rest
```

File: components/Button/index.tsx [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
 }: Props) {
   return (
     <button
```
Comment (Darguima) [inline components/Button/index.tsx:20]:
```suggestion
    <button
      {...rest}
```

**Review by Darguima — COMMENTED** [review]
I would suggest take the `...rest` of the Button Props and pass it to the button element, to allow to use the generic HTML button props at the component level, without need future changes or extra complexity.

I think that if you don't want do this, at least you should correct the Props types with the Typescript Pick or Omit operators, because at this moment is correct, at the typescript level, pass any of the HTML buttons props to this component, even if they aren't used.

image

Also, if you want accept the changes I think that would be better the `customStyle` be the `className` (already included at ...rest) so as not to create confusion, since they would overwrite the other

File: components/Button/index.tsx [hunk 5]
```diff
@@ -1,27 +1,44 @@
...
 interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
-  text: string;
+  title: string;
+  description?: string;
   customStyle?: string;
+  bold?: boolean;
+}
+
+interface ButtonTitleProps {
+  title: string;
```
Comment (Darguima) [inline components/Button/index.tsx:11]: Change this type to `ReactNode`

File: components/Button/index.tsx [hunk 6]
```diff
@@ -1,27 +1,44 @@
 import { ButtonHTMLAttributes } from "react";
 
 interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
-  text: string;
+  title: string;
```
Comment (Darguima) [inline components/Button/index.tsx:4]: Change this type to `ReactNode`

**Review by Darguima — CHANGES_REQUESTED** [review]

**Review by Darguima — DISMISSED** [review]

**Review by Darguima — APPROVED** [review]
Perfect

**Pull request merged**